tests/fastapi/test_context.py

```diff
@@ -41,11 +84,43 @@ def abc(self, info: Info) -> str:
 
     app = FastAPI()
     schema = strawberry.Schema(query=Query)
-    graphql_app = GraphQLRouter(schema)
+    graphql_app = GraphQLRouter(schema, context_getter=None)
     app.include_router(graphql_app, prefix="/graphql")
 
     test_client = TestClient(app)
     response = test_client.post("/graphql", json={"query": "{ abc }"})
 
     assert response.status_code == 200
     assert response.json() == {"data": {"abc": "abc"}}
+
+
+def test_with_invalid_context_getter():
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def abc(self, info: Info) -> str:
+            assert info.context.get("request") is not None
+            assert info.context.get("strawberry") is None
+            return "abc"
+
+    def custom_context_dependency() -> str:
+        return "rocks"
+
+    def get_context(value: str = Depends(custom_context_dependency)) -> str:
+        return value
+
+    app = FastAPI()
+    schema = strawberry.Schema(query=Query)
+    graphql_app = GraphQLRouter(schema=schema, context_getter=get_context)
+    app.include_router(graphql_app, prefix="/graphql")
+
+    test_client = TestClient(app)
+    try:
+        response = test_client.post("/graphql", json={"query": "{ abc }"})
+    except TypeError as e:
+        assert (
+            str(e) == "The custom context dependency must be either a "
+            "class that inherits from BaseContext or a dictionary"
+        )
+        response = "Task failed successfully"
+    assert response == "Task failed successfully"
```

Comment:
```suggestion
    assert response == "Task failed successfully"

```
Also, missing new line at end of file

Reply:
this should have been covered by precommit 🤔